### DISCUSSION TITLE
Fetch: connection closed before message completed

### DISCUSSION THREAD

**Discussion by retog** [denoland/deno#13070] [Q&A] [answered]
My application executes many fetch requests. Even if all `await fetch` calls are in try/catch blocks execution terminates with the message:
```
error: Uncaught (in promise) TypeError: error sending request for url (http://fuseki:3330/ds/update): connection closed before message completed
    at async mainFetch (deno:ext/fetch/26_fetch.js:266:14)
Process exited with code 1
```
The same message seems to appear also when serving HTTP: https://github.com/denoland/deno/blob/5c0636888c7b8686fca671a48718c7cf11ef252f/cli/tests/unit/http_test.ts#L153, https://github.com/denoland/deno/issues/11595

I've tried adding delays between fetches as well as setting keepalive to false. Any idea on how to avoid this error? 

**Update**

Turns out I did miss catching the rejections of a promise. Catching the exception prevents the program from terminating.

**Comment by lucacasonato** [comment, marked as answer]
For future reference, an update from OP:

> Turns out I did miss catching the rejections of a promise. Catching the exception prevents the program from terminating.
